packages/cli/src/commands/integration/add.ts

```diff
@@ -1,86 +1,284 @@
-import { Team } from '@vercel-internals/types';
+import chalk from 'chalk';
 import open from 'open';
 import Client from '../../util/client';
-import getUser from '../../util/get-user';
+import formatTable from '../../util/format-table';
+import { packageName } from '../../util/pkg-name';
+import getScope from '../../util/get-scope';
+import list from '../../util/input/list';
+import cmd from '../../util/output/cmd';
+import indent from '../../util/output/indent';
 import { getLinkedProject } from '../../util/projects/link';
-import getTeamById from '../../util/teams/get-team-by-id';
-
-export interface IntegrationProduct {
-  id: string;
-  slug: string;
-  name: string;
-  shortDescription: string;
-  type: 'storage' | string;
-}
-
-export interface Integration {
-  id: string;
-  slug: string;
-  name: string;
-  products?: IntegrationProduct[];
-}
+import {
+  BillingPlan,
+  Integration,
+  IntegrationProduct,
+  Metadata,
+} from './types';
+import { createMetadataWizard } from './wizard';
 
 export async function add(client: Client, args: string[]) {
   if (args.length > 1) {
     client.output.error(`Can't install more than one integration at a time.`);
     return 1;
   }
 
+  const { contextName, team } = await getScope(client);
+
+  if (!team) {
+    client.output.error('Team not found');
+    return 1;
+  }
+
   const integrationSlug = args[0];
+  const integration = await fetchIntegration(client, integrationSlug);
+
+  if (!integration?.products) {
+    return 1;
+  }
 
-  const user = await getUser(client);
-  const teamId = client.config.currentTeam ?? user.defaultTeamId;
+  const [product, installations] = await Promise.all([
+    selectProduct(client, integration),
+    fetchInstallations(client, integration),
+  ]);
 
-  if (!teamId) {
-    client.output.error('Team not found');
+  if (!product || !installations) {
+    return 1;
+  }
+
+  const installation = installations.find(
+    install =>
+      install.ownerId === team.id && install.installationType === 'marketplace'
+  );
+
+  client.output.log(
+    `Installing ${chalk.bold(product.name)} by ${chalk.bold(integration.name)} under ${chalk.bold(contextName)}`
+  );
+
+  const metadataSchema = product.metadataSchema;
+  const metadataWizard = createMetadataWizard(metadataSchema);
+
+  if (
+    !installation ||
+    !metadataWizard.isSupported ||
+    product.type !== 'storage'
```

Comment:
Added some comments in the code. TL;DR we only have the necessary APIs for storage atm. If don't make this check and we add another category to the Marketplace the CLI would break.

We will eventually add generic APIs so this check isn't necessary anymore.